cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -412,6 +408,13 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 				a.clusterStateRegistry.Recalculate()
 			}
 
+			if (scaleDownStatus.Result == status.ScaleDownNoNodeDeleted ||
+				scaleDownStatus.Result == status.ScaleDownInCooldown ||
```

Comment:
We are already in branch where ScaleDownInCooldown and ScaleDownInProgress did not happen.
If you decide that it should be moved outside of the branch, make sure that it would not remove soft taints after node deletion starts.

Reply:
You're completely right. Removing these conditions from the check to keep the change minimal.